langchain4j-core/src/main/java/dev/langchain4j/web/search/WebSearchPagination.java

```diff
@@ -0,0 +1,129 @@
+package dev.langchain4j.web.search;
+
+import java.util.Map;
+import java.util.Objects;
+
+import static dev.langchain4j.internal.ValidationUtils.ensureGreaterThanZero;
+
+
+/**
+ * Represents pagination information for web search results. Normally it's used as part of a {@link WebSearchResults} to navigate through the search results.
+ * Typically, this includes the current page number, the URL of the next page, the URL of the previous page,
+ * and a map of page numbers to their respective URLs.
+ */
+public class WebSearchPagination {
```

Comment:
Is this really required? For which use cases will it be useful?
If not, I would rather remove it (less code = less bugs = less work etc)